bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/E4HandlerProxy.java

```diff
@@ -50,7 +52,7 @@
  * @since 3.5
  *
  */
-public class E4HandlerProxy implements IHandler2, IHandlerListener, IElementUpdater {
+public class E4HandlerProxy implements IHandler2, IHandlerListener, IElementUpdater, IObjectWithState {
```

Comment:
Since it is interpreted as an API breakage, requiring an API filter; couldn't we instead encapsulate the state as a package-visible final field ?

Reply:
The API is detected to be extended, not broken. I found that reducing visibility is impossible. Here is the chain of thought:

1. `org.eclipse.e4.core.commands.internal.HandlerServiceHandler` is used by Workbench to hydrate Command Framework
1. `HandlerServiceHandler` works with  `org.eclipse.e4.core.contexts.IEclipseContext`  available to consumers of Command Framework
1. User can populate `IEclipseContext` with their own handlers using `org.eclipse.e4.core.commands.EHandlerService`
1. User expects the protocols to match those used by Command and Workbench frameworks
1. HandlerServiceHandler has to detect the workbench protocols in the provided handlers to stay compatible with user's handlers. If other protocols are in use an additional different logic is required to detect them, which I deem unacceptable.
1. Therefore Workbench handlers have to use same protocols as user's handlers

Reply:
On a second thought, this is an internal API and @iloveeclipse says, API tools should not complain about extensions of internal APIs. I guess, I'll try to revert relevant version bumps.

Reply:
Relevant version bumps were reverted previously. `org.eclipse.ui.workbench` has original version, which indicates that API tools have not found any API changes at all.